autoload/db.vim

```diff
@@ -310,17 +342,25 @@ function! db#execute_command(mods, bang, line1, line2, cmd) abort
       if exists('lines')
         call writefile(lines, infile)
       endif
-      call s:filter_write(conn, infile, outfile)
       execute 'autocmd BufReadPost' fnameescape(tr(outfile, '\', '/'))
             \ 'let b:db_input =' string(infile)
             \ '| let b:db =' string(conn)
             \ '| let w:db = b:db'
             \ '| call s:init()'
+      execute 'autocmd BufUnload' fnameescape(tr(outfile, '\', '/'))
+            \ 'call db#job#cancel()'
       let s:results[conn] = outfile
+
+      if !s:async
+        call s:filter_write(conn, infile, outfile)
+      else
+        call writefile(['Running query...'], outfile)
```

Comment:
Without even trying it, I can tell this will look bad in file types with strict syntax highlighting such as JSON. Can we get away with just an `:echo`?

One thing I have been toying with for a similar feature in another plugin is doing `:set modified` in the buffer, which makes it _feel_ like a work in progress, and furthermore synergizes nicely with job cancellation, as now you need to do `:q!` to kill a job but just `:q` to close after it's finished.

Reply:
I added it because the preview window opens a file that does not exist at that moment. This was just to ensure there's something to open. I could write an empty file and let it populate asynchronously, and do what you suggested.

Reply:
Oh, yeah, empty file is the way to go then.